tests/common.go

```diff
@@ -13948,6 +13949,76 @@ func GetVolumeNamefromPVC(namespace string) ([]string, error) {
 	return nil, fmt.Errorf("No PVCs found in namespace [%s]", namespace)
 }
 
+func DoVolumeQuorumValidation(volumeQuorumValidationStopSignal chan struct{}, vQourumError *error) {
+	itr := 1
+	for {
+		select {
+		case <-volumeQuorumValidationStopSignal:
+			log.Infof("Exiting volume quorum validation routine")
+			return
+		default:
+			log.Infof("Volume quorum validation iteration: #%d", itr)
+			errorChan := make(chan error, 50)
+			ValidateVolumeQuorum(&errorChan)
+			close(errorChan) // Close the channel to signal completion
+			for err := range errorChan {
+				if err != nil {
+					*vQourumError = multierr.Append(*vQourumError, err)
+				}
+			}
+			if *vQourumError != nil {
+				return
+			}
+			itr++
+			time.Sleep(10 * time.Second)
+		}
+	}
+}
+
+func ValidateVolumeQuorum(errChan ...*chan error) {
+	volIDs, err := Inst().V.ListAllVolumes()
+	if err != nil {
+		// add to errChan
+		err := fmt.Errorf("error listing volumes, Err: %v", err)
+		processError(err, errChan...)
+		return
+	}
+
+	for _, volID := range volIDs {
+		apiVol, err := Inst().V.InspectVolume(volID)
+		if err != nil {
+			err = fmt.Errorf("error inspecting volume [%s], Err: %v", volID, err)
+			processError(err, errChan...)
+			return
+		}
+		// check if volume is up
+		log.Infof("Volume [%s] status: %v", volID, apiVol.Status)
+		if apiVol.Status != opsapi.VolumeStatus_VOLUME_STATUS_UP {
```

Comment:
@nikita-bhatia We don't expose if a volume is in resync or not from our SDK. I believe the volume will show up as STATUS_UP when resync is running. As we don't expose this information from the SDK, we will have to call `pxctl volume inspect` and look at that data to determine if a volume is in quorum and in sync.

Reply:
In a repl-3 volume, it is seen that even when one of 3 node is being updated, volume inspect shows replication status in "Degraded" state, is that something we want to consider here?
<img width="1607" alt="image" src="https://github.com/user-attachments/assets/51d4c3b2-6b0e-4edb-a336-7348e28559fe">

Reply:
As discussed with @piyush-nimbalkar , have added extra validation to validate Runtime sttaus of the volumes along with volume status. This is going to validate that nodes which are down for upgrades are in "clean" RunTime state only.